app/assets/javascripts/plate-map/add-data-on-change.js

```diff
@@ -151,6 +178,18 @@ var plateLayOutWidget = plateLayOutWidget || {};
         var checkboxes = this.globalSelectedAttributes.slice(); 
         var selectedAreas = this.selectedAreas.slice(); 
         var focalWell = this.focalWell;
+        var selectedWells;
+
+        return {
+          "derivative": derivative,
+          "checkboxes": checkboxes,
+          "selectedAreas": selectedAreas,
+          "focalWell": focalWell
+        };
+      },
+
+      getCurrentPlate: function(){
+        var derivative = $.extend(true, {}, this.engine.derivative);
         var colorLocMap = {};
         var colorLocIdxMap = this.engine.stackUpWithColor;
         var dim = $("#my-plate-layout").plateLayOut("getDimensions");
```

Comment:
This only works if the layout object has the id my-plate-layout, which is not guaranteed. Should be using `var dim = this.getDimensions();`

Reply:
Also do `var that = this;` and line below line inside loop needs to be `return that.indexToAddress(locIdx, dim);`

Reply:
Agreed and changed!